pkg/testutils/lint/passes/deferunlockcheck/deferunlockcheck.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package deferunlockcheck
+
+import (
+	"go/ast"
+
+	"github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/passesutil"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+// Doc documents this pass.
+const Doc = "checks that usages of mutex Unlock() are deferred."
+
+const noLintName = "deferunlock"
+
+// Please try to avoid updating this value, use the nolint comment instead.
+const unlocksLimit = 1037
+
+// Analyzer is an analysis pass that checks for mutex unlocks which
+// aren't deferred.
+var Analyzer = &analysis.Analyzer{
+	Name:     "deferunlockcheck",
+	Doc:      Doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	astInspector := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	// Note: defer ...Unlock() expressions are captured as ast.DeferStmt nodes so we don't have to worry about those results returning
+	// for ast.ExprStmt nodes.
+	filter := []ast.Node{
+		(*ast.ExprStmt)(nil),
+	}
+	var unlocksFound int
+	astInspector.Preorder(filter, func(n ast.Node) {
+		stmt := n.(*ast.ExprStmt)
+		expr, ok := stmt.X.(*ast.CallExpr)
+		if !ok {
+			return
+		}
+		sel, ok := expr.Fun.(*ast.SelectorExpr)
+		if !ok {
+			return
+		}
+		// Increase the count if we found a violation.
+		if sel.Sel != nil && (sel.Sel.Name == "Unlock" || sel.Sel.Name == "RUnlock") && !passesutil.HasNolintComment(pass, n, noLintName) {
+			unlocksFound++
+		}
+		// If we exceed the limit then the reports trigger.
+		if unlocksFound > unlocksLimit {
```

Comment:
What is this supposed to do? If a given package has more than 1037 manual locks/unlocks it is an error? What is the significance of this number?